### PR TITLE
update is_most_recent_record in sub history

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
## PR Overview
**This PR will address the following Issue/Feature:**  Issue fivetran/dbt_recharge#38 

internal ticket where a customer was seeing a partition by constant expression issue due to a null value as the partition key resulting from having `id` in their source subscription_history table instead of `subscription_id`. To fix this we separated out the partition so that the coalesce between `id` and `subscription_id` occurred first.

**This PR will result in the following new package version:**

v0.3.1

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Bugfix
- Updates the `is_most_recent_record` field in `stg_recharge__subscription_history` to partition on `id`, or `subscription_id` if the former is not available in the source table.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [na] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Hex notebook linked in ticket!

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# dbt_recharge_source v0.3.1
+[PR #15](https://github.com/fivetran/dbt_recharge_source/pull/15) includes the following updates:
+## Bugfix
+- Updates the `is_most_recent_record` field in `stg_recharge__subscription_history` to partition on `id`, or `subscription_id` if the former is not available in the source table.
```
Comment (fivetran-catfritz) [inline CHANGELOG.md]: I would add an explanation of why this had to be done.

**Review by fivetran-catfritz — COMMENTED** [review]
@fivetran-reneeli Code changes look good! Suggestion for the changelog, and also left a comment in Height re the validation. Thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (fivetran-reneeli) [reply CHANGELOG.md]: Added!

**Review by fivetran-catfritz — APPROVED** [review]
Awesome--lgtm!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
Thanks @fivetran-reneeli for this quick fix. These changes look good to me!

**Pull request merged**